terraform/aws/modules/elasticache/elasticache.tf

```diff
@@ -0,0 +1,122 @@
+
+resource "aws_elasticache_cluster" "redis_node" {
+    cluster_id                     = "${var.environment}-${var.name}-node"
+    engine                         = var.engine
+    engine_version                 = var.redis_version
+    node_type                      = var.instance_type
+    num_cache_nodes                = var.cluster_size
+    parameter_group_name           = aws_elasticache_parameter_group.cache_parameter_group.id
+    subnet_group_name              = "${aws_elasticache_subnet_group.cache_subnet_group.id}"
+    security_group_ids             = [aws_security_group.cache_sg.id]
+    maintenance_window             = var.maintenance_window
+    apply_immediately              = var.apply_immediately
+    notification_topic_arn         = var.notification_topic_arn
+    port                           = var.port
+    az_mode                        = var.az_mode
+    availability_zone              = var.availability_zone
+    preferred_availability_zones   = var.availability_zones
+    count                          = var.engine == "redis" && var.replication_group_enabled == "disabled" ? 1 : 0
+}
+
+# Create Redis cache cluster
+
+resource "aws_elasticache_replication_group" "redis" {
+  replication_group_id          = "${var.environment}-${var.name}"
+  description                   = "Terraform-managed ElastiCache replication group for ${var.environment}-${var.name}"
+  num_cache_clusters            = var.cluster_size
+  node_type                     = var.instance_type
+  automatic_failover_enabled    = var.redis_failover
+  auto_minor_version_upgrade    = var.auto_minor_version_upgrade
+  preferred_cache_cluster_azs   = var.availability_zones
+  multi_az_enabled              = var.multi_az_enabled
+  engine                        = var.engine
+  at_rest_encryption_enabled    = var.at_rest_encryption_enabled
+  kms_key_id                    = var.kms_key_id
+  transit_encryption_enabled    = var.transit_encryption_enabled
+  auth_token                    = var.transit_encryption_enabled ? var.auth_token : null
+  engine_version                = var.redis_version
+  port                          = var.port
+  parameter_group_name          = aws_elasticache_parameter_group.cache_parameter_group.id
+  subnet_group_name             = aws_elasticache_subnet_group.cache_subnet_group.id
+  security_group_names          = var.security_group_names
+  security_group_ids            = [aws_security_group.cache_sg.id]
+  snapshot_arns                 = var.snapshot_arns
+  snapshot_name                 = var.snapshot_name
+  apply_immediately             = var.apply_immediately
+  maintenance_window            = var.maintenance_window
+  notification_topic_arn        = var.notification_topic_arn
+  snapshot_window               = var.redis_snapshot_window
+  snapshot_retention_limit      = var.redis_snapshot_retention_limit
+  count                         = var.engine == "redis" && var.replication_group_enabled == "enabled" ? 1 : 0
```

Comment:
looks good. Can you please move the count to the top of the resource. It makes it more readable.

Reply:
Done